### DISCUSSION TITLE
Simplified syntax highlight

### DISCUSSION THREAD

**Discussion by fserb** [zed-industries/zed#8859] [Feature Requests]
I appreciate the tree-sitter approach that Zed has. It has extremely good performance and good coverage.

That said, it would be really nice if there was a mechanism to create syntax highlight for less common languages that was simplified (regexps of other regexp groups, like [textmate and vscode support](https://manual.macromates.com/en/language_grammars).

For common languages, the current system has a great performance/difficulty tradeoff. It's okey to spend a bunch of time describing C/C++/Javascript/Rust/etc... But for one-off niche things, tree-sitter (CFG + hacky C code) is a bit overkill and the wrong solution to the problem. For those cases ("I have a special configuration file...."), something that is much simpler (even if less performant) is probably the right thing.

**Comment by matubu** [comment]
You are not required to write hacky C code for example I think the [zig tree-sitter](https://github.com/GrayJack/tree-sitter-zig/blob/master/grammar.js) only use the `grammar.js` to define the parser.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
